### ISSUE TITLE
Only one GPU is used during .fit() validation phase

### ISSUE THREAD

**Issue by netw0rkf10w** [tensorflow/tensorflow#34139]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- TensorFlow version (use command below): 2.0.0

**Describe the current behavior**

During validation phase of Keras' `.fit()`, only one GPU is used (while all GPUs are used during the training phase).

**Code to reproduce the issue**
This issue can be easily reproduced using the code of [this official tutorial](https://www.tensorflow.org/tutorials/distribute/keras), by feeding the validation dataset `eval_dataset` to `model.fit()`.

**Comment by guptapriya** [comment]
How did you determine if was using only one GPU during validation? Can you also share more information about your setup? This is using MirroredStrategy?

**Comment by guptapriya** [comment]
cc @qlzh727

**Comment by netw0rkf10w** [comment]
@guptapriya I used `nvidia-smi` to observe GPU usage, only one GPU is used during the entire validation phase of `.fit()`. Training was launched on 4 GPUs using `tf.distribute.MirroredStrategy()`. 

Not sure if it's related to this issue on Keras' repo: https://github.com/keras-team/keras/issues/9404.

**Comment by crccw** [comment]
from device placement logs it seems that validation happens on all GPUs under conditions mentioned in  keras-team/keras#9404 (num_data=7,num_gpu=3,batch_size=6).

nvidia-smi may not be accurate if the dataset is small. Could you try again with a larger validation set (by adding a .repeat(N) to the dataset).

**Comment by crccw** [comment]
Closing inactive issues. Feel free to reopen.

**Comment by keunwoochoi** [comment]
I seem to have the same problem. TF 2.4, keras. tf.data. Zero usage according to nvidia-smi and the validation procedure is obviously very slow.
